huaweicloud/services/ces/resource_huaweicloud_ces_alarmrule.go

```diff
@@ -377,112 +378,135 @@ func resourceConditionHash(v interface{}) int {
 	return hashcode.String(buf.String())
 }
 
-func buildAlarmAction(d *schema.ResourceData, name string) []alarmrule.ActionOpts {
-	if v, ok := d.GetOk(name); ok {
-		actionOptsRaw := v.([]interface{})
-		actionOpts := make([]alarmrule.ActionOpts, len(actionOptsRaw))
-		for i, actionOptRaw := range actionOptsRaw {
-			actionOpt := actionOptRaw.(map[string]interface{})
+func buildCreateAndUpdateAlarmActionV1BodyParams(rawParams interface{}) []map[string]interface{} {
+	if rawArray, ok := rawParams.([]interface{}); ok {
+		if len(rawArray) == 0 {
+			return nil
+		}
 
-			notificationListRaw := actionOpt["notification_list"].([]interface{})
+		params := make([]map[string]interface{}, len(rawArray))
+		for i, v := range rawArray {
+			raw := v.(map[string]interface{})
+			notificationListRaw := raw["notification_list"].([]interface{})
 			notificationList := make([]string, len(notificationListRaw))
 			for j, notification := range notificationListRaw {
 				notificationList[j] = notification.(string)
 			}
+			params[i] = map[string]interface{}{
+				"type":             raw["type"],
+				"notificationList": notificationList,
+			}
```

Comment:
no need to change into string list for json body
```suggestion
			params[i] = map[string]interface{}{
				"type":             raw["type"],
				"notificationList": raw["notification_list"],
			}
```